javascript-practice/src/js/views/homeView.js

```diff
@@ -0,0 +1,99 @@
+import CommonView from './commonView';
+
+export default class HomeView extends CommonView {
+  constructor() {
+    super();
+
+    this.tabs = document.querySelectorAll('.app__tab-item');
+    this.allContent = document.querySelectorAll('.app__content-item');
+    this.addTransactionBtn = document.getElementById('addTransaction');
+    this.addBudgetBtn = document.getElementById('addBudget');
+    this.dialogs = document.querySelectorAll('.dialog');
+    this.cancelBtn = document.querySelectorAll('.form__cancel-btn');
+    this.categoryField = document.getElementById('selectCategory');
+    this.closeIcon = document.querySelector('.close-icon');
+
+    this.budgetDialog = document.getElementById('budgetDialog');
+    this.transactionDialog = document.getElementById('transactionDialog');
+    this.categoryDialog = document.getElementById('categoryDialog');
+  }
+
+  /**
+   * Handle event when click on tabs
+   */
+  handlerTabsTransfer() {
+    this.tabs.forEach((tab, index) => {
+      tab.addEventListener('click', () => {
+        this.removeActiveTab();
+        tab.classList.add('active');
+
+        const line = document.querySelector('.app__line');
+
+        if (line.classList.contains('left')) {
+          HomeView.replaceClassElement('left', 'right', line);
+        } else {
+          HomeView.replaceClassElement('right', 'left', line);
+        }
+
+        this.allContent.forEach((content) => {
+          content.classList.remove('active');
+        });
+        this.allContent[index].classList.add('active');
+      });
+    });
+  }
+
+  static replaceClassElement(oldEl, newEl, el) {
```

Comment:
Not good for reuse, consider another way

Reply:
Fixed.